tests/test_fits.py

```diff
@@ -79,5 +80,31 @@ def test_fits_reading():
     im = imageio.imread(compressed)
     assert im.shape == (2042, 3054)
 
+@pytest.mark.skipif("astropy is None")
+def test_fits_get_reader():
+    """Test reading fits with get_reader method"""
+    
+    #Set precedence as normal
+    teardown_module()
```

Comment:
Could you explain why we have to tear down any module-level setup to run this test?

Reply:
Ah, I addressed this point on the `tmp_path` additions. The setup changes the precedence from normal to being default FITS format, so my test was always passing. I noticed the `teardown` function and decided to take advantage of it to set precedence levels back to normal, where the original issue occurs.

Reply:
In the most recent commit I removed the `teardown_module` call and I just use the `imageio.formats.sort()` line to return the format precedence to normal for the test.

Reply:
I see, thanks for the explanation. `teardown_module` seems like a very big hammer for this. 

In another pending PR we are using the module-scope to download and generate a bunch of test images that can then easily be copied into a local folder for a test to consume. Tearing down the module may have unwanted side-effects here, such as forcing us to regenerate the files (which is expensive).

`imageio.formats.sort()` might also persist after the test has finished, since `imageio.formats` is a singleton and (I think) that libraries aren't reloaded between tests. I'll dig into this tomorrow and see if further changes are needed.

Reply:
Okay, I went through the code. Both `teardown_module` and `imageio.formats.sort()` won't do the trick here, because they will both leave the environment in the wrong state after the test completes.

Instead, we need a new fixture that _temporarily_ reverts the order back to normal. It will look something like this

```python
@pytest.fixture
def normal_plugin_order():
    teardown_module()
    yield
    setup_module()
```

This will run the test using the normal plugin order and revert to the module's default state after the test is finished. It should be a fixture because that will take care of most corner cases for us, e.g. if the test crashes with an exception the environment will still be restored correctly.

We have to use that fixture by updating the test's signature `def test_fits_get_reader(normal_plugin_order, tmp_path):`.

@astrophysaxist Is this explanation sufficient to describe what has to be modified?

Reply:
Yep that makes perfect sense @FirefoxMetzger . Happy to see pytest has this capability. Learned a lot about pytest functionality with this! Just committed those changes.